x-pack/platform/plugins/shared/actions/common/validate_email_addresses.ts

```diff
@@ -14,8 +14,20 @@ import { hasMustacheTemplate } from './mustache_template';
 export interface ValidateEmailAddressesOptions {
   /** treat any address which contains a mustache template as valid */
   treatMustacheTemplatesAsValid?: boolean;
+  // addresses with this option won't be validated against the allowed recipient list
+  isSender?: boolean;
 }
 
+export const isAddressMatchingSomePattern = (address: string, patterns: string[]): boolean => {
+  return patterns.some((pattern) => {
+    // replaces '.' with '\.' and '*' with '.*' for regex matching
+    const regexStr = '^' + pattern.replace(/\./g, '\\.').replace(/\*/g, '.*') + '$';
```

Comment:
This doesn't feel safe enough, since it only escapes _some_  RegExp chars.

I did notice the lodash function [`escapeRegExp()`](https://lodash.com/docs/4.17.15#escapeRegExp).  You could imagine doing a `split()` on `*`, then using `escapeRegExp()` on all the pieces, then do a `join('*') on the result.  

Hard to imagine we don't have something like this in Kibana already, if `minimatch` is that expensive.

Reply:
done in 487962ab9ee947d8dc532dadcbd3d2e8a37430ff